NullReferenceException in UrlUtils.EncryptParameter

## Description of bug
Evoq runs the File Crawler search provider (a Scheduler task) to index file contents. This provider is based on the `StandardFolderProvider` class. We are seeing following error now:

![image](https://user-images.githubusercontent.com/30123437/84443126-665ec900-ac15-11ea-96ea-a5ee3ba97088.png)

This is actually a `NullReferenceException` when the Search API calls the `StandardFolderProvider.GetFileUrl(IFileInfo)` method:

![image](https://user-images.githubusercontent.com/30123437/84442168-7d9cb700-ac13-11ea-845f-baefde60d2de.png)

This happens because the `EncryptParameter(string)` method calls the `PortalController.Instance.GetCurrentPortalSettings()` method, which is known to give a NRE in the context of a Scheduler task (as in #2606).

![image](https://user-images.githubusercontent.com/30123437/84442431-fb60c280-ac13-11ea-8232-d5dcf5e3b7a1.png)

In this particular case, we can avoid this issue by calling the `EncryptParameter(string, string)` overload (passing the encryption key), which we can obtain safely by using the `PortalSettings` instance available in `GetFileUrl(IFileInfo)` (line 178 in first screenshot).

## Steps to reproduce
The steps to reproduce are simply to run the "Search: File Crawler" task, which is specific to Evoq, but actually any search provider based on the `StandardFolderProvider` will crash.

## Error information
```
2020-06-11 17:35:40.353-03:00 [DNN-DEV-02][D:9][T:73][ERROR] DotNetNuke.Services.FileSystem.FileManager - System.NullReferenceException: Object reference not set to an instance of an object.
   at DotNetNuke.Common.Utilities.UrlUtils.EncryptParameter(String value)
   at DotNetNuke.Services.FileSystem.StandardFolderProvider.GetFileUrl(IFileInfo file)
   at DotNetNuke.Services.FileSystem.FileManager.GetUrl(IFileInfo file)
2020-06-11 17:35:40.376-03:00 [DNN-DEV-02][D:9][T:73][ERROR] DotNetNuke.Services.Exceptions.Exceptions - DotNetNuke.Services.Exceptions.BasePortalException: File Crawler Unable to Finish, Error: The underlying system threw an exception. ---> DotNetNuke.Services.FileSystem.FolderProviderException: The underlying system threw an exception. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at DotNetNuke.Common.Utilities.UrlUtils.EncryptParameter(String value)
   at DotNetNuke.Services.FileSystem.StandardFolderProvider.GetFileUrl(IFileInfo file)
   at DotNetNuke.Services.FileSystem.FileManager.GetUrl(IFileInfo file)
   --- End of inner exception stack trace ---
   at DotNetNuke.Services.FileSystem.FileManager.GetUrl(IFileInfo file)
   at DotNetNuke.Professional.SearchCrawler.Results.FileResultController.GetDocUrl(SearchResult searchResult)
   at DotNetNuke.Services.Search.Controllers.SearchControllerImpl.GetSecurityTrimmedResults(SearchQuery searchQuery, LuceneQuery luceneQuery)
   at DotNetNuke.Services.Search.Controllers.SearchControllerImpl.GetResults(SearchQuery searchQuery)
   at DotNetNuke.Services.Search.Controllers.SearchControllerImpl.SiteSearch(SearchQuery searchQuery)
   at DotNetNuke.Professional.SearchCrawler.FileCrawling.FileCrawlerController.GetLastIndexedFileModificationDate(IFolderInfo folder)
   at DotNetNuke.Professional.SearchCrawler.FileCrawling.FileCrawlerController.IndexDirectory(ISearchCrawlerPortalContext searchCrawlerPortalContext, IFolderInfo folderInfo)
   at DotNetNuke.Professional.SearchCrawler.FileCrawling.FileCrawlerController.IndexPortalDirectory(ISearchCrawlerPortalContext searchCrawlerPortalContext)
   at DotNetNuke.Professional.SearchCrawler.FileCrawling.FileCrawlerController.IndexPortalDirectories(ISearchCrawlerContext searchCrawlerContext)
   at DotNetNuke.Professional.SearchCrawler.FileCrawler.FileCrawler.FileCrawlerExecution()
   --- End of inner exception stack trace ---
```

## Affected version
* [X] 10.00.00 alpha build (in development)
* [X] 09.06.01 latest supported release

--- a/DNN Platform/Library/Services/FileSystem/Providers/StandardFolderProvider.cs
+++ b/DNN Platform/Library/Services/FileSystem/Providers/StandardFolderProvider.cs
@@ -199,7 +199,9 @@ namespace DotNetNuke.Services.FileSystem
             // Does site management want the cachebuster parameter?
             if (portalSettings.AddCachebusterToResourceUris)
             {
-                var cachebusterToken = UrlUtils.EncryptParameter(file.LastModificationTime.GetHashCode().ToString());
+                var cachebusterToken = UrlUtils.EncryptParameter(
+                    file.LastModificationTime.GetHashCode().ToString(),
+                    portalSettings.GUID.ToString());
 
                 return TestableGlobals.Instance.ResolveUrl(fullPath + "?ver=" + cachebusterToken);
             }

--- a/DNN Platform/Tests/DotNetNuke.Tests.Core/Providers/Folder/StandardFolderProviderTests.cs
+++ b/DNN Platform/Tests/DotNetNuke.Tests.Core/Providers/Folder/StandardFolderProviderTests.cs
@@ -604,6 +604,36 @@ namespace DotNetNuke.Tests.Core.Providers.Folder
         #region GetFilesUrl
 
         [Test]
+        public void GetFileUrl_WhenCurrentPortalSettingsReturnsNull_DontThrow()
+        {
+            // arrange
+            var sfp = new Mock<StandardFolderProvider>
+            {
+                CallBase = true
+            };
+
+            sfp.Setup(x => x.GetPortalSettings(Constants.CONTENT_ValidPortalId))
+                .Returns(GetPortalSettingsMock());
+
+            _fileInfo.Setup(x => x.FileName)
+                .Returns(Constants.FOLDER_ValidFileName);
+
+            _fileInfo.Setup(x => x.PortalId)
+                .Returns(Constants.CONTENT_ValidPortalId);
+
+            _portalControllerMock.Setup(x => x.GetCurrentPortalSettings())
+                .Returns<PortalSettings>(null);
+
+            // act
+            string fileUrl = null;
+            TestDelegate action = () => fileUrl = sfp.Object.GetFileUrl(_fileInfo.Object);
+
+            // assert
+            Assert.DoesNotThrow(action);
+            Assert.IsNotNull(fileUrl);
+        }
+
+        [Test]
         [TestCase("(")]
         [TestCase(")")]
         [TestCase("")]        

